spectator-reg-atlas/src/main/java/com/netflix/spectator/atlas/impl/QueryIndex.java

```diff
@@ -230,7 +244,19 @@ private boolean remove(List<Query.KeyQuery> queries, int i, T value) {
             if (idx.isEmpty())
               equalChecks.remove(v);
           }
+        } else if (kq instanceof Query.Has) {
+          if (hasKeyIdx != null) {
+            result |= hasKeyIdx.remove(queries, j, value);
+            if (hasKeyIdx.isEmpty())
+              hasKeyIdx = null;
+          }
         } else {
+          if (kq instanceof Query.Regex) {
+            Query.Regex re = (Query.Regex) kq;
+            otherChecksTree.remove(re.pattern().prefix(), kq);
```

Comment:
Need `result |= ` on this and L258?

Reply:
It shouldn't be needed, the result is updated based on the removal from the otherChecks map.